internal/sink/sink.go

```diff
@@ -745,12 +754,13 @@ func (v2 *WebhookV2) updateUrls(urlCount int, url string, urls []string) {
 
 // worker is the routine that actually takes the queued messages and delivers
 // them to the listeners outside webpa
-func (v2 *WebhookV2) send(msg *wrp.Message) error {
+func (v2 *WebhookV2) send(msg *wrp.Message, wg *sync.WaitGroup) error {
 	defer func() {
 		if r := recover(); nil != r {
 			// s.DropsDueToPanic.With(prometheus.Labels{metrics.UrlLabel: s.id}).Add(1.0)
 			v2.logger.Error("goroutine send() panicked", zap.String("id", v2.id), zap.Any("panic", r))
 		}
+		wg.Done()
 		// s.workers.Release()
```

Comment:
Instead of creating `wg *sync.WaitGroup` for every `batchSend`, we could leverage something like `WebhookV2`'s `s.workers.Release()`. But it'll live in `WebhookSink` instead. This would act as a rate limiter, making sure 1 Sink is not using all the server's resources. It doesn't have to be a semaphore, a channel can provide the same functionality